pkg/kubelet/cm/cpumanager/state/state_file.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"encoding/json"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"os"
+	"sync"
+)
+
+type stateFileData struct {
+	DefaultCPUSet string            `json:"defaultCpuSet"`
+	Entries       map[string]string `json:"reservedList,omitempty"`
+}
+
+type stateFile struct {
+	sync.RWMutex
+	stateFilePath string
+	State
```

Comment:
Let's make this a regular member instead of embedding the interface. That way if we add functions to the `state.State` interface later, the compiler will help us remember to implement them for this type instead of inheriting implementations from the memory state. Maybe call the member `cache`?

Reply:
ok, can do, i just wanted to avoid implementing all `Get` methods